Code/HighLevelAPI/src/main/java/ie/tcd/wayfinder/highlevel/navigation/controller/NavigationController.java

```diff
@@ -80,9 +88,12 @@ public String index()
         return new ResponseEntity<String>(responseString, HttpStatus.valueOf(response.getStatusLine().getStatusCode()));
     }
     
-    @GetMapping("/navigation/start/{start}/destination/{dest}/{mode}/{username}")
-    public ResponseEntity<String> findRoute(@PathVariable String start, @PathVariable String dest, @PathVariable TravelMode mode, @PathVariable String username) throws IOException
+    @CrossOrigin(origins = "*")
+    @GetMapping("/navigation/start/{start}/destination/{dest}/{mode}")
+    @PreAuthorize("hasAuthority('ROLE_USER')")
+    public ResponseEntity<String> findRoute(@AuthenticationPrincipal Principal principal, @PathVariable String start, @PathVariable String dest, @PathVariable TravelMode mode) throws IOException
     {
+    	String username = principal.getName();
     	
     	System.out.println("\n\n\n GOT REQUEST \n\n\n");
```

Comment:
Instead of sysout, do you want to log it?

Reply:
That's is not my code, haha

Reply:
haha